common/predictive-text/worker/models/common.ts

```diff
@@ -0,0 +1,25 @@
+namespace models {
+  export class Common {
+    static applyTransform(transform: Transform, context: Context): Context {
+      // First, get the current context
+      let fullLeftContext = context.left || '';
+      let lLen = fullLeftContext.length;
+      let lDel = lLen < transform.deleteLeft ? lLen : transform.deleteLeft;
+
+      let leftContext = fullLeftContext.substr(0, lLen - lDel) + (transform.insert || '');
```

Comment:
```suggestion
      let leftContext = fullLeftContext.substring(0, lLen - lDel) + (transform.insert || '');
```

`.substr()` is a [legacy function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr), and its use is discouraged. That's why VSCode autosuggests it when you start typing "myString.subs" ;)